includes/AdaptiveThumb.php

```diff
@@ -46,10 +46,19 @@ public static function pichtml($input, array $argv, Parser $parser, PPFrame $fra
     $alt = htmlentities($argv['alt']);
     $margin = htmlentities($argv['margin']);
     
-    if (!empty($link)) {
-      $linkopen="<a href=$link>"; 
-      $linkclose="</a>";
+    $prefix = "";
+    
+    // media viewer needs a file in a page to be triggered, include an invisible file tag
+    if (!empty($fileName) && empty($link)) {
+        // class="metadata": don't display this image itself (otherwise it will appear twice in media viewer
+        $dummyMarkup = "<span class=\"metadata\" style=\"display:none\">[[File:$fileName]]</span>";
+        $prefix = $parser->recursiveTagParse($dummyMarkup, $frame);
     }
+    
+    // either a trigger for the media viewer or a custom link
+    $linkopen = empty($link) ?  "<a class=\"image\" style=\"cursor:pointer\">" : "<a href=$link>";
```

Comment:
well, this line is not compatible to guys who do not do coding every day. While I understand your good intention here, I would like to have this out of my code. Anyway, will take it over until I provide a "better" (i.e. more readable to sporadic programmers) code.